Core/src/org/sleuthkit/autopsy/corecomponents/DataContentViewerHTML.java

```diff
@@ -0,0 +1,293 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.corecomponents;
+
+import org.sleuthkit.autopsy.modules.filetypeid.FileTypeDetector;
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.util.logging.Level;
+import com.sun.javafx.application.PlatformImpl;
+import javafx.stage.Stage;
+import javafx.embed.swing.JFXPanel;
+import javafx.event.ActionEvent;
+import javafx.geometry.Insets;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.web.WebView;
+import javafx.scene.web.WebEngine;
+import org.openide.util.NbBundle;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.openide.nodes.Node;
+import org.openide.util.Exceptions;
+import org.openide.util.lookup.ServiceProvider;
+import org.sleuthkit.autopsy.corecomponentinterfaces.DataContentViewer;
+import org.sleuthkit.autopsy.modules.filetypeid.FileTypeDetector.FileTypeDetectorInitException;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskData.TSK_DB_FILES_TYPE_ENUM;
+
+
+
+/**
+ * HTML view of file contents. This viewer uses a JavaFX WebView to display HTML data.
+ */
+@ServiceProvider(service = DataContentViewer.class, position = 6)
+public class DataContentViewerHTML extends javax.swing.JPanel implements DataContentViewer {
+
+    private AbstractFile dataSource;
+    private FileTypeDetector fileTypeDetector;
+    private Stage stage;
+    private WebView browser;
+    private JFXPanel jfxPanel;
+    private WebEngine webEngine;
+    private Scene scene;
+    private Button javaScriptButton;
+    
+    private final String enableJavaScriptTxt;
+    private final String disableJavaScriptTxt;
+    
+    private static final Logger logger = Logger.getLogger(DataContentViewerHTML.class.getName());
+    
+  
+    public DataContentViewerHTML() {
+        
+        try {
+            fileTypeDetector = new FileTypeDetector();
+        } catch (FileTypeDetectorInitException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+        
+        initComponents();
+        resetComponents();
+        
+        enableJavaScriptTxt = NbBundle.getMessage(this.getClass(), 
+                "DataContentViewerHTML.enableJavaScriptBtnTxt");
+        disableJavaScriptTxt = NbBundle.getMessage(this.getClass(), 
+                "DataContentViewerHTML.disableJavaScriptBtnTxt");        
+        
+        
+        logger.log(Level.INFO, "Created HTMLView instance: " + this); //NON-NLS
+    }
+    
+    private void initComponents() {     
+        
```

Comment:
The precedent of using initComponents is a legacy of previous practices.  I would recommend in-lining this in the constructor, probably also in-lining the createWebScene method.  @rcordovano , what do you think?